### PR TITLE
Lesson 13 adrian filimon

### PR DESCRIPTION
**Pull request by AdrianFilimon1707**

### CODE REVIEW

File: lesson_13/adrian_filimon/lesson_13.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+class PerimeterMixin:
+    def perimeter(self):
+
+        class_name = type(self).__name__
+
+        return f"This is the perimeter for {class_name}: {sum(self.sides)}"
+
+
+class Polygon(PerimeterMixin):
```
Comment (lorena91) [inline lesson_13/adrian_filimon/lesson_13.py:11]: I think that the requirement was to make PerimeterMixin available only for triangle class so it should be passed as an argument only for the triangle class

File: lesson_13/adrian_filimon/lesson_13.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+class Square(Polygon):
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        side, *_ = self.sides
+        return side ** 2
+
+    def from_area(area):
```
Comment (Liitle) [inline lesson_13/adrian_filimon/lesson_13.py]: This method is supposed to return a new Square object. The logic of getting the sides is correct, but with the sides, we need to create a new object. In order to create a new object we need the class object (cls).

What we can use from python to achieve this?

**Review by Liitle — CHANGES_REQUESTED** [review]

File: lesson_13/adrian_filimon/lesson_13.py (hunk 1 above)
Comment (AdrianFilimon1707) [reply lesson_13/adrian_filimon/lesson_13.py:11]: Oh yes.I can change that so it better fits the requirement,  but I wanted to give all the classes that functionality and practice multiple inheritance.

File: lesson_13/adrian_filimon/lesson_13.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        super().__init__(*args)
+
+    def area(self):
+        side, *_ = self.sides
+        return side ** 2
+
+    @classmethod
+    def from_area(cls, area):
+        sq = sqrt(area)
+        return f"Length of side is {sq}"
```
Comment (Liitle) [inline lesson_13/adrian_filimon/lesson_13.py]: You added the class method but the return is still a string. it is better to return an instance. `cls(sq,sq,sq,sq)`

**Review by Liitle — CHANGES_REQUESTED** [review]

File: lesson_13/adrian_filimon/lesson_13.py (hunk 3 above)
Comment (Liitle) [reply lesson_13/adrian_filimon/lesson_13.py]: @AdrianFilimon1707 pinging you just as a reminder to take a look at this PR :smile: :smile: :smile:

**Comment by AdrianFilimon1707** [comment]
Didn't get the notification. sorry
I will check again

**Review by Liitle — APPROVED** [review]

**Pull request merged**